tests/telemetry/test_telemetry.py

```diff
@@ -258,3 +281,236 @@ def test_virtualdb_table_streaming(duthosts, rand_one_dut_hostname, ptfhost, loc
+                ids=["monit_config_line"])
+def test_mem_spike_setup_and_cleanup(duthosts, rand_one_dut_hostname, setup_streaming_telemetry, request):
+    """Customizes Monit configuration files before testing and restores them after testing.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        setup_streaming_telemetry: Fixture to setup telemetry server authentication before testing.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    backup_monit_config_files(duthost)
+    customize_monit_config_files(duthost, request.param)
+    restart_monit_service(duthost)
+
+    yield
+
+    restore_monit_config_files(duthost)
+    restart_monit_service(duthost)
+
+
+@pytest.mark.disable_loganalyzer
+def test_mem_spike(duthosts, rand_one_dut_hostname, ptfhost, test_mem_spike_setup_and_cleanup, gnxi_path):
+    """Test whether memory usage of telemetry container will increase and be restarted
+    or not by Monit if python gNMI client continuously creates channels with gNMI server.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        pfthost: PTF docker binding to the selected DuT.
+        test_mem_spike_setup_and_cleanup: Fixture does testing setup and cleanup.
+
+    Returns:
+        None.
+    """
+    logger.info("Starting to test the memory spike issue of '{}' ...".format(CONTAINER_NAME))
+
+    duthost = duthosts[rand_one_dut_hostname]
+    dut_ip = duthost.mgmt_ip
+
+    logger.info("Checking whether the '{}' container is running before testing...".format(CONTAINER_NAME))
+    is_running = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                            CONTAINER_CHECK_INTERVAL_SECS,
+                            0,
+                            check_container_state, duthost, CONTAINER_NAME, True)
+    pytest_assert(is_running, "'{}' is not running on DuT!".format(CONTAINER_NAME))
+    logger.info("'{}' is running on DuT!".format(CONTAINER_NAME))
+
+    expected_alerting_messages = []
+    expected_alerting_messages.append(".*restart_service.*Restarting service 'telemetry'.*")
+    expected_alerting_messages.append(".*Stopping Telemetry container.*")
+    expected_alerting_messages.append(".*Stopped Telemetry container.*")
+    expected_alerting_messages.append(".*Starting Telemetry container.*")
+    expected_alerting_messages.append(".*Started Telemetry container.*")
+
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="container_restart_due_to_memory")
+    loganalyzer.expect_regex = []
+    loganalyzer.expect_regex.extend(expected_alerting_messages)
+    marker = loganalyzer.init()
+
+    client_thread = ThreadPool(processes=1)
+    client_thread.apply_async(run_gnmi_client, (ptfhost, dut_ip, gnxi_path, -1))
```

Comment:
>-1 [](http://example.com/codeflow?start=76&length=2)

You are using inf connection. For a fixed image, it will finally lead to client side OOM. Suggest use a large number which could OOM on bad image, but pass good image.